app/process/watcher/ethereum/watcher.go

```diff
@@ -28,44 +29,59 @@ import (
 type Watcher struct {
 	config    config.Ethereum
 	contracts service.Contracts
+	ethClient client.Ethereum
 	logger    *log.Entry
 }
 
-func NewWatcher(contracts service.Contracts, config config.Ethereum) *Watcher {
+func NewWatcher(contracts service.Contracts, ethClient client.Ethereum, config config.Ethereum) *Watcher {
 	return &Watcher{
 		config:    config,
 		contracts: contracts,
+		ethClient: ethClient,
 		logger:    c.GetLoggerFor("Ethereum Watcher"),
 	}
 }
 
 func (ew *Watcher) Watch(queue *pair.Queue) {
 	go ew.listenForEvents(queue)
-	log.Infof("Listening for events at contract [%s]", ew.config.RouterContractAddress)
+	ew.logger.Infof("Listening for events at contract [%s]", ew.config.RouterContractAddress)
 }
 
 func (ew *Watcher) listenForEvents(q *pair.Queue) {
 	events := make(chan *routerContract.RouterBurn)
 	sub, err := ew.contracts.WatchBurnEventLogs(nil, events)
 	if err != nil {
-		log.Errorf("Failed to subscribe for Burn Event Logs for contract address [%s]. Error [%s].", ew.config.RouterContractAddress, err)
+		ew.logger.Errorf("Failed to subscribe for Burn Event Logs for contract address [%s]. Error [%s].", ew.config.RouterContractAddress, err)
 	}
 
 	for {
 		select {
 		case err := <-sub.Err():
-			log.Errorf("Burn Event Logs subscription failed. Error: [%s].", err)
+			ew.logger.Errorf("Burn Event Logs subscription failed. Error: [%s].", err)
 			return
 		case eventLog := <-events:
-			ew.handleLog(eventLog, q)
+			go ew.handleLog(eventLog, q)
 		}
 	}
 }
 
 func (ew *Watcher) handleLog(eventLog *routerContract.RouterBurn, q *pair.Queue) {
-	log.Infof("New Burn Event Log for [%s], Amount [%s], Receiver Address [%s] has been found.",
+	ew.logger.Infof("New Burn Event Log for [%s], Amount [%s], Receiver Address [%s] has been found.",
```

Comment:
Maybe it would be best if we use this log after we have the block confirmations. Instead of the line 83 log. 
Here we can add log that is `debug` level just so that we know that event was received, but after the block confirmations are done we should have the normal `info` log.

Proposal for this log:
`ew.logger.Debugf("New Burn Event Log received. Waiting block confirmations")` <- we could include TX has maybe as-well